File: examples/ai/quickstart.py
import os

from sdk.python.copperbench import CopperbenchClient, read_workspace_connection

workspace_path = os.environ["COPPERBENCH_WORKSPACE"]
connection = read_workspace_connection(workspace_path)
client = CopperbenchClient(
    os.environ.get("COPPERBENCH_MCP_URL", connection["url"]),
    os.environ["COPPERBENCH_TOKEN"],
    os.environ.get("COPPERBENCH_WORKSPACE_ID", connection["workspaceId"]),
)
client.initialize()
workspace = client.get_workspace()
revision = workspace.get("data", {}).get("workspace", {}).get("revision", 0)
for element in client.list_mod_elements(fields=["id", "name", "type"]):
    print(element)
print(client.build_workspace(revision))

File: examples/ai/quickstart.ts
import { CopperbenchClient, readWorkspaceConnection } from '../../sdk/typescript/copperbench.js';

const workspacePath = process.env.COPPERBENCH_WORKSPACE ?? '';
if (!workspacePath) throw new Error('COPPERBENCH_WORKSPACE must point to the workspace folder or .mcreator file');
const connection = readWorkspaceConnection(workspacePath);
const client = new CopperbenchClient({
  endpoint: process.env.COPPERBENCH_MCP_URL ?? connection.url,
  token: process.env.COPPERBENCH_TOKEN ?? '',
  workspaceId: process.env.COPPERBENCH_WORKSPACE_ID ?? connection.workspaceId
});

await client.initialize();
const workspace = await client.getWorkspace();
const revision = Number((workspace.data as { workspace?: { revision?: number } })?.workspace?.revision ?? 0);
for await (const element of client.listModElements({ fields: ['id', 'name', 'type'] })) {
  console.log(element);
}
console.log(await client.buildWorkspace(revision));
